services/ui-src/src/utils/state/useStore.ts

```diff
@@ -0,0 +1,129 @@
+import { create } from "zustand";
+import { devtools, persist } from "zustand/middleware";
+// types
+import {
+  McrUserState,
+  MCRUser,
+  AdminBannerData,
+  AdminBannerState,
+  ReportShape,
+  McrReportState,
+  ReportMetadataShape,
+  EntityShape,
+  McrEntityState,
+} from "types";
+
+// USER STORE
+const userStore = (set: Function) => ({
+  // initial state
+  user: undefined,
+  // show local logins
+  showLocalLogins: undefined,
+  // actions
+  setUser: (newUser?: MCRUser) =>
+    set(() => ({ user: newUser }), false, { type: "setUser" }),
+  // toggle show local logins (dev only)
+  setShowLocalLogins: () =>
+    set(() => ({ showLocalLogins: true }), false, { type: "showLocalLogins" }),
+});
+
+// BANNER STORE
+const bannerStore = (set: Function) => ({
+  // initial state
+  bannerData: undefined,
+  bannerActive: false,
+  bannerLoading: false,
+  bannerErrorMessage: "",
+  bannerDeleting: false,
+  // actions
+  setBannerData: (newBanner: AdminBannerData | undefined) =>
+    set(() => ({ bannerData: newBanner }), false, { type: "setBannerData" }),
```

Comment:
I was briefly looking at this with Ben, and he brought up a question about the type declarations here. Are we being unnecessarily verbose with the declared types, or is this necessary to thread through the debugging of actions we desire?

Reply:
I think it's fine. It would always be nicer to have less code, but the debuggability is probably worth it here.